### PR TITLE
V0.12.1.x govobj submission

### PR DESCRIPTION
**Pull request by evan82**
Fixed governance object submission issues
    
    - Terrible bug which caused governance objects to have their hashes corrupted, causing objects to become invalid
    - More proposal -> governance name changes

### CODE REVIEW

**Comment by UdjinM6** [comment]
Good find! 👍 
Needs rebase

File: src/governance.cpp [hunk 1]
```diff
@@ -28,8 +28,9 @@ std::vector<CGovernanceObject> vecImmatureGovernanceObjects;
 
 int nSubmittedFinalBudget;
 
-bool IsCollateralValid(uint256 nTxCollateralHash, uint256 nExpectedHash, std::string& strError, int64_t& nTime, int& nConf, CAmount minFee)
+bool IsCollateralValid(uint256 nTxCollateralHash, uint256 nExpectedHash, std::string& strError, int64_t nTime, int& nConf, CAmount minFee)
```
Comment (UdjinM6) [inline src/governance.cpp]: On another though: what's the point of passing nTime here at all?

**Comment by UdjinM6** [comment]
@schinzelh why this was merged? My note wasn't addressed in any way yet and compiler gives a warning.
```
governance.cpp: In function ‘bool IsCollateralValid(uint256, uint256, std::string&, int64_t, int&, CAmount)’:
governance.cpp:31:6: warning: parameter ‘nTime’ set but not used [-Wunused-but-set-parameter]
 bool IsCollateralValid(uint256 nTxCollateralHash, uint256 nExpectedHash, std::string& strError, int64_t nTime, int& nConf, CAmount minFee)
      ^
At global scope:
cc1plus: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
```

**Comment by schinzelh** [comment]
@UdjinM6 My bad, I'll revert it

File: src/governance.cpp (hunk 1 above)
Comment (evan82) [inline src/governance.cpp]: Wow, you're right. I'll fix it.

**Comment by evan82** [comment]
I've resolved the other issues by removing nTime from IsCollateralValid. It seems to be an artifact of the old way the system used to retrieve nTime for the tx.

**Pull request merged**